superbench/runner/runner.py

```diff
@@ -143,17 +144,17 @@ def __get_mode_command(self, benchmark_name, mode, timeout=None):
                 'mpirun '    # use default OpenMPI in image
                 '-tag-output '    # tag mpi output with [jobid,rank]<stdout/stderr> prefix
                 '-allow-run-as-root '    # allow mpirun to run when executed by root user
-                '{host_list} '    # use prepared hostfile and launch {proc_num} processes on each node
+                '{host_list} '    # use prepared hostfile or specify nodes and launch {proc_num} processes on each node
                 '-bind-to numa '    # bind processes to numa
                 '{mca_list} {env_list} {command}'
             ).format(
-                host_list=f'-host localhost:{mode.proc_num}'
-                if mode.node_num == 1 else f'-hostfile hostfile -map-by ppr:{mode.proc_num}:node',
+                host_list=f'-host localhost:{mode.proc_num}' if mode.node_num == 1 else \
+                    ('-hostfile hostfile' if mode.host_list is None else '-host ' + ','.join(
+                        f'{host}:{mode.proc_num}' for host in mode.host_list)) + f' -map-by ppr:{mode.proc_num}:node',
```

Comment:
seems the last `-map-by` is only required for `-hostfile`, `'{host}:{mode.proc_num}'` is sufficient for `-host`

Reply:
Tested on cluster and verified you are right, Will address it.